ARrehab/ARrehab/TileGrid.swift

```diff
@@ -0,0 +1,163 @@
+    */
+    func generatePossibleTiles() {
+        let xExtent = self.surfaceAnchor.extent.x
+        let zExtent = self.surfaceAnchor.extent.z
+        
+        let xSize = Tile.TILE_SIZE.x
+        let zSize = Tile.TILE_SIZE.z
+        
+        var currentX = xExtent/2
+        var currentZ = zExtent/2
+        
+        //Generate the tiles
+        while abs(currentX) <= xExtent/2 {
+            while abs(currentZ) <= zExtent/2 {
+                self.generateOneTile(currentX: currentX, currentZ: currentZ)
+                currentZ -= zSize
+            }
+            currentZ = zExtent/2
+            currentX -= xSize
+        }
+        
+        self.xLength = getMaxX().0.x
+        self.zLength = getMaxZ().0.z
+    }
+    
+    /*
+     Generates a tile with the given coordinates
+     Adds the tile to the possibleTiles dict. and to the gridEntity
+     */
+    func generateOneTile(currentX: Float, currentZ: Float) {
+        let newTile = Tile(name: String(format: "Tile (%f,%f)", currentX, currentZ), x: currentX, z: currentZ, materials: [TileGrid.gridMaterial], adjustTranslation: true)
+        self.possibleTiles[newTile.coords] = newTile
+        self.gridEntity.addChild(newTile)
+    }
+    
+    /*
+     Updates the tile grid if the plane has expanded sufficiently
+     */
+    func updateBoard(updatedAnc: ARPlaneAnchor) {
+        //Update the stored anchor
+        self.surfaceAnchor = updatedAnc
+        
+        /*
+         Check if the new plane is long enough to fit more tiles
+         If so, remove all current tiles and regenerate grid to fit
+         Lastly, adjust translation so grid is re-centered correctly
+         */
+        if (self.surfaceAnchor.extent.x > self.xLength * 2 + Tile.TILE_SIZE.x || self.surfaceAnchor.extent.z > self.zLength * 2 + Tile.TILE_SIZE.z) {
+            
+            self.gridEntity.children.removeAll()
+            self.generatePossibleTiles()
+            
+            self.gridEntity.transform.translation = self.surfaceAnchor.center
+        }
+    }
+    
+    //TODO -- IMPLEMENT DIRECTIONAL KNOWLEDGE
+    func updateBoardOutline(centerTile: Tile) {
+        
+        //Clear current outline
+        self.clearOutline()
+        
+        //Find the tile that would form the upper-right corner of the board, with the given centerTile
+        let cornerCoords = Tile.Coordinates(x: centerTile.coords.x + (1 * Tile.TILE_SIZE.x), z: centerTile.coords.z + (2 * Tile.TILE_SIZE.z))
+        
+        let cornerTile = self.possibleTiles.first() {coords, tile in
+            return TileGrid.isApproxEqual(value1: coords.x, value2: cornerCoords.x, error: 0.1) && TileGrid.isApproxEqual(value1: coords.z, value2: cornerCoords.z, error: 0.1)
+        }
+        
+        if cornerTile == nil {
+            print("Invalid center")
+            return
+        }
+        
+        else {
+            for i in 0...2 {
```

Comment:
Is `i` and `c` hard coded versions of the board dimensions? If so, is there a way we can use change these values?

Reply:
Yes and yes